experiments/benchmarks/land.jl

```diff
@@ -0,0 +1,671 @@
+        retention_model = retention_model,
+    )
+    plant_hydraulics_args = (
+        parameters = plant_hydraulics_ps,
+        n_stem = n_stem,
+        n_leaf = n_leaf,
+        compartment_midpoints = compartment_midpoints,
+        compartment_surfaces = compartment_surfaces,
+    )
+
+    energy_args = (parameters = Canopy.BigLeafEnergyParameters{FT}(ac_canopy),)
+
+    # Canopy component args
+    canopy_component_args = (;
+        autotrophic_respiration = autotrophic_respiration_args,
+        radiative_transfer = radiative_transfer_args,
+        photosynthesis = photosynthesis_args,
+        conductance = conductance_args,
+        hydraulics = plant_hydraulics_args,
+        energy = energy_args,
+    )
+
+    # Other info needed
+    shared_params = Canopy.SharedCanopyParameters{FT, typeof(earth_param_set)}(
+        z0_m,
+        z0_b,
+        earth_param_set,
+    )
+
+    canopy_model_args = (;
+        parameters = shared_params,
+        domain = ClimaLand.obtain_surface_domain(domain),
+    )
+
+    # Integrated plant hydraulics and soil model
+    land_input = (atmos = atmos, radiation = radiation)
+    land = SoilCanopyModel{FT}(;
+        soilco2_type = soilco2_type,
+        soilco2_args = soilco2_args,
+        land_args = land_input,
+        soil_model_type = soil_model_type,
+        soil_args = soil_args,
+        canopy_component_types = canopy_component_types,
+        canopy_component_args = canopy_component_args,
+        canopy_model_args = canopy_model_args,
+    )
+
+    Y, p, cds = initialize(land)
+
+    init_soil(ν, θ_r) = θ_r + (ν - θ_r) / 2
+    Y.soil.ϑ_l .= init_soil.(ν, θ_r)
+    Y.soil.θ_i .= FT(0.0)
+    T = FT(276.85)
+    ρc_s =
+        Soil.volumetric_heat_capacity.(
+            Y.soil.ϑ_l,
+            Y.soil.θ_i,
+            soil_params.ρc_ds,
+            soil_params.earth_param_set,
+        )
+    Y.soil.ρe_int .=
+        Soil.volumetric_internal_energy.(
+            Y.soil.θ_i,
+            ρc_s,
+            T,
+            soil_params.earth_param_set,
+        )
+    Y.soilco2.C .= FT(0.000412) # set to atmospheric co2, mol co2 per mol air
+    Y.canopy.hydraulics.ϑ_l.:1 .= plant_ν
+    evaluate!(Y.canopy.energy.T, atmos.T, t0)
+
+    set_initial_cache! = make_set_initial_cache(land)
+    exp_tendency! = make_exp_tendency(land)
+    imp_tendency! = ClimaLand.make_imp_tendency(land)
+    tendency_jacobian! = ClimaLand.make_tendency_jacobian(land)
+    set_initial_cache!(p, Y, t0)
+
+    # set up jacobian info
+    jac_kwargs = (;
+        jac_prototype = ImplicitEquationJacobian(Y),
+        Wfact = tendency_jacobian!,
+    )
+
+    prob = SciMLBase.ODEProblem(
+        CTS.ClimaODEFunction(
+            T_exp! = exp_tendency!,
+            T_imp! = SciMLBase.ODEFunction(imp_tendency!; jac_kwargs...),
+            dss! = ClimaLand.dss!,
+        ),
+        Y,
+        (t0, tf),
+        p,
+    )
+
+    updateat = Array(t0:(3600 * 3):tf)
```

Comment:
do we want this to be in terms of `dt` or `tf`? It doesn't have to be, but it is for the other benchmarks

Reply:
I thought we could do every 3 hours because that is what we had discussed doing in real simulations (or at most, every hour). I think we had left another benchmark as updating every timestep, and that is helpful for working on optimizing the updating/file IO code, but I dont know that we need to do that in all files.